Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -813,12 +813,11 @@ static std::optional<TimeZoneRecord> parseTimeZone(StringParsingBuffer<Character
 }
 
 template<typename CharacterType>
-static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<CharacterType>& buffer)
+static std::optional<CalendarRecord> parseOneCalendar(StringParsingBuffer<CharacterType>& buffer)
 {
     // For BNF, see comment in canBeCalendar()
 
-    if (!canBeCalendar(buffer))
-        return std::nullopt;
+    ASSERT(canBeCalendar(buffer));
```

Comment:
I wonder if this should be a release assert, or perhaps we can keep the existing nullopt return case, just in case this can somehow be reached.

Reply:
In that case, maybe it's easiest to drop the assertion and keep a nullopt return. In this function it's trivial to prove the assertion (`parseOneCalendar` is called only in a `while(canBeCalendar(buffer))` loop in `parseCalendar`) but the other site below is less straightforward to prove.